docs: Add hosting setup how-to (#422)
unoplatform/uno.extensions: doc/How-Tos/Hosting/Hosting-setup.md

# How-To: Get Started with Hosting

> Hosting can be used to register services that will be accessible throughout the application via dependency injection. This tutorial will walk you through the critical pieces needed to leverage hosting in your application. 

Reviewer: ```suggestion
**Uno.Extensions.Hosting** can be used to register services that will be accessible throughout the application via dependency injection (DI). This tutorial will walk you through the critical pieces needed to leverage hosting in your application.
```

> NOTE: If you used the `dotnet new` template to create your solution, the steps outlined here are unecessary. You can see whether your application already creates the IHost interface using the CreateDefaultBuilder method by looking for a generated `App.xaml.host.cs` file.

Reviewer: ```suggestion
NOTE: If you used the `dotnet new unoapp-extensions` template to create your solution, the steps outlined here are unnecessary. You can see whether your application already creates the `IHost` interface using the `CreateDefaultBuilder` method by looking for a generated `App.xaml.host.cs` file.
```

## Step-by-steps
1. Install the Uno.Extensions.Hosting library from NuGet, making sure you reference the package that is appropriate for the flavor of Uno in use. Likewise, projects with Uno.WinUI require installation of [Microsoft.Extensions.Hosting.WinUI](https://www.nuget.org/packages/Uno.Extensions.Hosting.WinUI) while those with Uno.UI need [Microsoft.Extensions.Hosting.UWP](https://www.nuget.org/packages/Uno.Extensions.Hosting.UWP) instead.

> 2. Use the UnoHost static class to create IHost for the application when the application instance is created:

Reviewer: ```suggestion
2. Use the `UnoHost` static class to create the `IHost` for the application when the application instance is created:
```

    * We need to expose the IHost instance to the rest of the application. Add the following property to your App.xaml.cs class:
        ```cs
        public IHost Host { get; private set; }
        ```

>     * Invoke the CreateDefaultBuilder() method in UnoHost followed by Build() 

Reviewer: ```suggestion
    * Invoke the `CreateDefaultBuilder()` method in `UnoHost` followed by `Build()`
```

# How to get started with hosting

Hosting can be used to register services that will be accessible throughout the application via dependency injection. This tutorial will walk you through the critical pieces needed to leverage hosting in your application. 

NOTE: If you used the `dotnet new` template to create the solution, some of the steps outlined here are unecessary. You can see whether your application already creates the IHost interface using the CreateDefaultBuilder method by looking for a generated `App.xaml.host.cs` file.

## Step-by-steps

> 1. Install the Uno.Extensions.Hosting library from NuGet, making sure you reference the package that is appropriate for the flavor of Uno in use.

Reviewer 2: Be specific here as to which packages to install - don't forget some readers will be completely new and won't understand what you mean, so will try to install both packages

> 2. Set the LangVersion to leverage new C# features

Reviewer 2: Remove this - it was included in my blog because I wanted to use implicit usings and some other c#10 features but not necessary here I think
